sdk/search/azure-search/src/main/java/com/azure/search/models/OcrSkill.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+
+/**
+ * A skill that extracts text from image files.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@odata.type")
+@JsonTypeName("#Microsoft.Skills.Vision.OcrSkill")
+@Fluent
+public final class OcrSkill extends Skill {
+    /*
+     * A value indicating which algorithm to use for extracting text. Default
+     * is printed. Possible values include: 'printed', 'handwritten'
+     */
+    @JsonProperty(value = "textExtractionAlgorithm")
+    private TextExtractionAlgorithm textExtractionAlgorithm;
```

Comment:
This property is deprecated and will now default to null.  The backend will completely ignore but it still exists on the skill model for backwards compatibility purposes.  Any idea what we should do with it here to not promote it anymore?

Reply:
@careyjmac I don't -- the new API guidelines didn't mention this scenario the last time I checked.

@JonathanGiles Any suggestions here, other than to just mention in the documentation string that this property is deprecated?

Reply:
@brjohnstmsft we can add a depercated annotation to the class when generating the code . Will require an update to the autorest config in the rest repo

Reply:
@navalev @JonathanGiles Is that an acceptable approach according to the Azure API guidelines?

Reply:
@brjohnstmsft I don't see anything specific in the guidelines.

Reply:
@navalev Ok, let's not block the PR on this. If you can provide an example offline of the change we'd need to make, we can take it from there.